File: code/pipeline-cpp/src/config/args.cpp
#include "args.h"

Args::Args(const int argc, const char* const* argv)
{
  argparse::ArgumentParser parser(argv[0]);

  parser.add_argument("-s", "--settings")
    .help("The name of the settings file")
    .default_value(std::string("settings.toml"));

  parser.parse_args(argc, argv);
  settings_ = parser.get<std::string>("--settings");
}

File: code/pipeline-py/config/args.py
import argparse
from dataclasses import dataclass


@dataclass
class Args:
    """Parses command line arguments for the application.

    If the `--settings` argument is not provided, it defaults to
    'settings.toml'."""

    settings: str
    """The name of the settings file."""

    def __init__(self):
        """Constructs an Args object and parses command line arguments.

        Initialises an ArgumentParser and parses the provided arguments."""

        parser = argparse.ArgumentParser(description="Python pipeline.")

        parser.add_argument(
            '-s',
            '--settings',
            type=str,
            default='settings.toml',
            help="Path to the settings TOML file",
        )

        args = parser.parse_args()
        self.settings = args.settings
